### ISSUE TITLE
Number type is not convertible to int/float

### ISSUE THREAD

**Issue by acampove** [microsoft/pyright#11093]
**Describe the bug**

I am trying to use `Number` from the standard library to type annotate numbers, but then when I try to cast to an actual number I see an error

**Code or Screenshots**

```python
from numbers import Number

def to_int(val: Number) -> int:
    return int(val)

print(to_int(3.5))
print(to_int(42))
```

which shows:

```
WARNING: there is a new pyright version available (v1.1.406 -> v1.1.407).
Please install the new version or set PYRIGHT_PYTHON_FORCE_VERSION to `latest`

/home/acampove/Tests/pyright/number.py
  /home/acampove/Tests/pyright/number.py:4:16 - error: Argument of type "Number" cannot be assigned to parameter "x" of type "ConvertibleToInt" in function "__new__"
    Type "Number" is not assignable to type "ConvertibleToInt"
      "Number" is not assignable to "str"
      "Number" is incompatible with protocol "Buffer"
        "__buffer__" is not present
      "Number" is incompatible with protocol "SupportsInt"
        "__int__" is not present
      "Number" is incompatible with protocol "SupportsIndex"
        "__index__" is not present
    ... (reportArgumentType)
  /home/acampove/Tests/pyright/number.py:4:16 - error: Argument of type "Number" cannot be assigned to parameter "x" of type "ConvertibleToInt" in function "__new__"
    Type "Number" is not assignable to type "ConvertibleToInt"
      "Number" is not assignable to "str"
      "Number" is incompatible with protocol "Buffer"
        "__buffer__" is not present
      "Number" is incompatible with protocol "SupportsInt"
        "__int__" is not present
      "Number" is incompatible with protocol "SupportsIndex"
        "__index__" is not present (reportArgumentType)
  /home/acampove/Tests/pyright/number.py:6:14 - error: Argument of type "float" cannot be assigned to parameter "val" of type "Number" in function "to_int"
    "float" is not assignable to "Number" (reportArgumentType)
  /home/acampove/Tests/pyright/number.py:7:14 - error: Argument of type "Literal[42]" cannot be assigned to parameter "val" of type "Number" in function "to_int"
    "Literal[42]" is not assignable to "Number" (reportArgumentType)
4 errors, 0 warnings, 0 informations
```

**Comment by erictraut** [comment]
Pyright is working correctly here. You'll see the same behavior from other Python type checkers like mypy. The `Number` class and `int` class are both nominal types with no overlap — that is, neither inherits from the other. The `Number` type should arguably be a protocol (a structural type definition), but it predates the invention of `Protocol` classes.
